### PR TITLE
Docstrings are all you need

### PR DESCRIPTION
**Pull request by vict0rsch**
# Description

This PR aims at providing at least a skeletton of docstrings everywhere. Wherever possible I'll try to fill them myself but in some cases it's either too complex or time-inefficient: the time it would take the function's author to explain things to me would be enough for them to actually write the docstring (hence: skeletton for them to fill-in the blanks)

* used existing docstrings where available
* created module / class / function-level docstrings (**Google-style**)
* wrapped executable code under `if __name__ == "__main__"` statement

## Type of change

- [x] New feature (non-breaking change which adds functionality) (not sure)
- [x] This change requires a documentation update (this is kind-of a circular reference)

# How Has This Been Tested?

No test as there's no code change as such

# Checklist:

- [ ] My code follows the style guidelines of this project (**are there any?**)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,12 +3,14 @@
 *.gif
 *.pkl
 *.json
-
+*.DS_Store
 # output files
 output/
+CTT-SHIPMENT-0
```
Comment (vict0rsch) [inline .gitignore:9]: Can this become an issue @mweiss17 ?

Comment (mweiss17) [reply .gitignore:9]: no this is fine

**Comment by vict0rsch** [comment]
(todo: replace `/!\` with Sphinx-compatible `rst` warnings)

**Comment by plstcharles** [comment]
@pg2455 could we get the magic admin finger to restart the tests on this (or... force-merge? it's documentation)

**Comment by pg2455** [comment]
How should I restart the tests?

**Pull request merged**